### PR TITLE
[Feature] - sc 29770 - open the piperider report in piperider run

### PR DESCRIPTION
**Pull request by jonycfu**
**PR checklist**

- [ X] Ensure you have added or ran the appropriate tests for your PR.
- [ X] DCO signed

**What type of PR is this?**
Feature
**What this PR does / why we need it**:
Provide opt-in flag `run --open` to open in browser upon report generation OR report uploading
**Which issue(s) this PR fixes**:
29770
**Special notes for your reviewer**:

### Cases covered:
```
# auto-upload: false

piperider run  # => no browser
piperider run --upload # => no browser
piperider run  --open => browser opens local file report
piperider run  --open --upload => browser opens uploaded file report on cloud

# auto-upload: true

piperider run  # => no browser
piperider run --upload # => no browser
piperider run  --open => browser opens uploaded file report on cloud
piperider run  --open --upload => browser opens uploaded file report on cloud


```
**Does this PR introduce a user-facing change?**: Yes

Added a new CLI option for `run` command, `--open`, which will auto open the report in your default browser depending on whether the report is uploaded to piperider cloud.

### CODE REVIEW

File: piperider_cli/generate_report.py [hunk 1]
```diff
@@ -122,3 +122,8 @@ def output_report(target_directory):
             console.print(f"Report generated in {output}/index.html")
         else:
             console.print(f"Report generated in {default_output_directory}/index.html")
+
+        # only open the local file report if auto-upload is OFF
+        if open_report and not auto_upload:
+            result_output = f"{output if output else default_output_directory}/index.html"
```
Comment (wcchang1115) [inline piperider_cli/generate_report.py:128]: browsers can't find the report when `output` is given a relative path

File: piperider_cli/generate_report.py [hunk 2]
```diff
@@ -122,3 +122,8 @@ def output_report(target_directory):
             console.print(f"Report generated in {output}/index.html")
         else:
             console.print(f"Report generated in {default_output_directory}/index.html")
+
+        # only open the local file report if auto-upload is OFF
+        if open_report and not auto_upload:
```
Comment (wcchang1115) [inline piperider_cli/generate_report.py:127]: browsers would open a local report and a cloud report when giving`--open` and `--upload` at the same time

**Review by wcchang1115 — CHANGES_REQUESTED** [review]
Hi Jon,
I found two unexpected behaviors as commented above
would you please take a look? thanks!

File: piperider_cli/generate_report.py (hunk 1 above)
Comment (jonycfu) [reply piperider_cli/generate_report.py:128]: how do we verify that it's a relative path vs. absolute path?

File: piperider_cli/generate_report.py (hunk 2 above)
Comment (jonycfu) [reply piperider_cli/generate_report.py:127]: Maybe I'm misreading this case, but it works on my side though 🤔 . 
` piperider run --open --upload  # this only opens in cloud on my local`

Based on my understanding, this statement should ONLY allow local report to open when either `--open` or `upload` is  `None`

File: piperider_cli/generate_report.py (hunk 1 above)
Comment (jonycfu) [reply piperider_cli/generate_report.py:128]: https://stackoverflow.com/questions/51520/how-to-get-an-absolute-file-path-in-python

File: piperider_cli/generate_report.py [hunk 3]
```diff
@@ -84,7 +84,7 @@ def _get_run_json_path(filesystem: FileSystem, input=None):
 
 class GenerateReport:
     @staticmethod
-    def exec(input=None, report_dir=None, output=None, open_report=None, auto_upload=None):
+    def exec(input=None, report_dir=None, output=None, open_report=None, open_in_cloud=False):
```
Comment (wcchang1115) [inline piperider_cli/generate_report.py:87]: seems that `open_report=None` can be `open_report=False` as well

**Review by wcchang1115 — APPROVED** [review]

**Pull request merged**